packages/calypso-e2e/src/lib/flows/gutenboarding-flow.ts

```diff
@@ -0,0 +1,205 @@
+import { Page } from 'playwright';
+import { getViewportName } from '../../browser-helper';
+
+export type Plans = 'Free' | 'Personal' | 'Premium' | 'Business' | 'eCommerce';
+export type Features =
+	| 'Custom domains'
+	| 'Store'
+	| 'SEO tools'
+	| 'Plugins'
+	| 'Ad-free'
+	| 'Image storage'
+	| 'Video storage'
+	| 'Priority support';
+
+const selectors = {
+	button: ( text: string ) => `button:text("${ text }")`,
+
+	// Start your website
+	siteTitle: '.acquire-intent-text-input__input',
+
+	// Domain
+	domainSearch: 'input[placeholder="Search for a domain"]',
+	domainSuggestionSpan: ( name: string, tld: string ) =>
+		`.domain-picker__suggestion-item-name span:has-text("${ name }${ tld }")`,
+
+	// Design
+	design: ( name: string ) => `button[data-e2e-button="freeOption"] span:text("${ name }")`,
+	fontPairingButton: ( platform: 'mobile' | 'desktop', fontName: string ) => {
+		return `.style-preview__font-options-${ platform } span:text("${ fontName }")`;
+	},
+	mobileFontPairingDropdown: `button.style-preview__font-option-select`,
+
+	// Features
+	featureItem: `button.features__item`,
+
+	// Plans
+	planItem: '.plans-accordion-item',
+	planName: '.plans-accordion-item__name',
+};
+
+/**
+ * Implements actions of the Gutenboarding process.
+ */
+export class GutenboardingFlow {
+	private page: Page;
+
+	/**
+	 * Constructs an instance of the component.
+	 *
+	 * @param {Page} page The underlying page.
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+	}
+
+	/**
+	 * Given a text, clicks on the first instance of the button with the text.
+	 *
+	 * @param {string} text User-visible text on the button.
+	 * @returns {Promise<void>} No return value.
+	 */
+	async clickButton( text: string ) {
+		await this.page.click( selectors.button( text ) );
+	}
+
+	/**
+	 * Enters the site title.
+	 *
+	 * This is equivalent to the Acquire Intent page from the Selenium suite.
+	 *
+	 * @param {string} title Title of the site.
+	 */
+	async enterSiteTitle( title: string ) {
+		await this.page.fill( selectors.siteTitle, title );
+	}
+
+	/**
+	 * Enters the keyword into the domain search input.
+	 *
+	 * @param {string} keyword Keyword used for domain search.
+	 */
+	async searchDomain( keyword: string ) {
+		await Promise.all( [
+			this.page.waitForSelector( '.is-placeholder', { state: 'hidden' } ),
+			this.page.fill( selectors.domainSearch, keyword ),
+		] );
+	}
+
+	/**
+	 * Given a domain name and TLD, select a matching domain.
+	 *
+	 * @param param0 Keyed object parameter.
+	 * @param {string} param0.name Name portion of the domain.
+	 * @param {string} param0.tld TLD portion of the domain.
+	 * @returns {Promise<string>} Selected domain.
+	 */
+	async selectDomain( { name, tld }: { name: string; tld: string } ): Promise< string > {
```

Comment:
Definite nit here: but I'm not entirely sure why we need to capture the name and tld as separate parameters here - it looks like we just concatenate them together later. In that case, would it just be better to have the user pass in exactly what domain they want to find? That way there's no risk they accidentally pass in "com" instead of ".com". In that case, we could also take out the code that returns the `innerText` of the found element - we either successfully find the exact URL they are looking for, or we don't!

Alternatively, I think capturing both separately makes a lot of sense if we want to do partial matching (like just a key word that needs to be available in the domain name). But in that case, we should probably use the [Playwright Regex based text selectors](https://playwright.dev/docs/selectors#text-selector) to find something that contains the keyword for the domain:

```typescript
	domainSuggestionSpan: ( name: string, tld: string ) =>
		`.domain-picker__suggestion-item-name :text-matches(".*${ name }.*\\.${ tld }", "i")`,
```

Reply:
This is a fair point and I had the following use case in mind when I did this:

1. user searches for `mycoolblog`
2. user selects `mycoolblog.com`.
3. user changes mind and returns to the domain search step.
4. user searches for `mysecondcoolblog.com`.

In this scenario, two domains with a `.com` TLDs are shown on page. 
I *think* I was trying to implement something flexible with this, but the end result didn't end up being like that. 
So I think it's preferable to default to a simpler implementation, to request user to enter the exact domain they would like.

Reply:
Fixed in 32357f3ed0c917646b87a95c963f9109d720b478.